### PR TITLE
refactor: Fix #442 Use map_ok in handlers to simplify the code and im…

### LINKED ISSUE #442: actix 2 cleanup: Change map to map_ok
based on feedback in https://github.com/mozilla-services/syncstorage-rs/pull/415#discussion_r382868153

### PR DESCRIPTION
**Pull request by fzzzy**
…prove error reporting.

## Description

Switching from future 0.1 to 0.3 changed the behavior of map. map_ok is closer to the old behavior and much more like what we really want.

## Testing

All tests should pass.

## Issue(s)

Closes #442.

### CODE REVIEW

File: src/web/handlers.rs [hunk 1]
```diff
@@ -418,15 +394,12 @@ pub fn get_bso(bso_req: BsoRequest) -> impl Future<Output = Result<HttpResponse,
...
-        .map(
-            |result: std::result::Result<std::option::Option<GetBso>, Error>| {
-                let result = result.expect("Could not get result in get_bso");
-                result.map_or_else(
-                    || Ok(HttpResponse::NotFound().finish()),
-                    |bso| Ok(HttpResponse::Ok().json(bso)),
-                )
-            },
-        )
+        .map_ok(|result: std::option::Option<GetBso>| {
```
Comment (pjenvey) [inline src/web/handlers.rs]:
```suggestion
        .map_ok(|result: Option<GetBso>| {
```

and also all the `std::collections::HashMap` -> `HashMap` (there's already a use statement for it)

Comment (pjenvey) [reply src/web/handlers.rs]: On second thought: 

```suggestion
        .map_ok(|result| {
```

..I doubt we need any of these type annotations any longer (the original version didn't)

File: src/web/handlers.rs [hunk 2]
```diff
@@ -187,12 +170,8 @@ where
...
         })
-        .map(
-            move |r: std::result::Result<
-                (Paginated<T>, std::result::Result<SyncTimestamp, Error>),
-                Error,
-            >| {
-                let (result, ts) = r.expect("Could not get r in finish_get_collection");
+        .map_ok(
+            move |(result, ts): (Paginated<T>, std::result::Result<SyncTimestamp, Error>)| {
                 let ts = ts.expect("Could not get ts in finish_get_collection");
```
Comment (pjenvey) [inline src/web/handlers.rs]: Forgot this last `expect()`

Comment (fzzzy) [reply src/web/handlers.rs]: No, I removed the expect for r but ts is still a result (both r and ts were results previously).

**Pull request merged**